### ISSUE TITLE
Offer compression for RRULE:BYDAY

### ISSUE THREAD

**Issue by dilyanpalauzov** [libical/libical#424]
`RRULE:FREQ=WEEKLY` and `RRULE:FREQ=WEEKLY,BYDAY=FR` are semantically equivallent (when DTSTART is on Friday, otherwise the interpretation is undefined).

Since the former form consumes less data, provide the possibility in libical to compress the RRULE in this case.

**Comment by ksmurchison** [comment]
Are you saying that you want libical to change the data that is handed to it by the application?
This can have implications for the application, especially a CalDAV server.  If the server changes the data that the client sends to  it, then it can not send an ETag in the response, meaning that the client has to fetch the resulting resource.  I don't see where it is libical's job to normalize RRULEs given to it by an application.

Also, libical currently CAN handle RRULE:FREQ=WEEKLY,BYDAY=FR with a DTSTART on a day other than FR.

**Comment by dilyanpalauzov** [comment]
My reading of RFC 5455 is, that when the repetitions in RRULE are on each Friday and DTSTART is on Wednesday, the interpretation is undefined.

I know that ETag breaks on changing the RRULE, but the ETag breaks also currently also when the server strips the VTIMEZONE provided alongside with VEVENT.  The CUA can use libical to compress the data before sending the data to the server.  Shouldn't the Schedule-Tag be used for such cases?  The point of doing this in libical is that all libical users get the functionality.

**Comment by ksmurchison** [comment]
Yes, RFC 5545 states that the behavior is undefined, but most users of libical expect the set of occurrences to start on the first FR on or after the day of week in DTSTART, so that is what has been implemented.

Schedule-Tag only applies to scheduling objects (those with an ORGANIZER property)

**Comment by dilyanpalauzov** [comment]
One more compression option: when the first instance of a repeating event is deleted, rather than inserting EXDATE, DTSTART shall be modified

libical shall in general offer an API that creates semantically equivalent iCalendar object, that is more normalized.
